File: VB/SimpleSchedulingExample/Doctor.vb
Public Class Doctor

    Public Shared Function Create() As Doctor
        Return New Doctor()
    End Function

    Public Shared Function Create(ByVal Id As Integer, ByVal Name As String) As Doctor
        Dim doctor As Doctor = Create()
        doctor.Id = Id
        doctor.Name = Name
        Return doctor
    End Function

    Protected Sub New()
    End Sub

    Public Overridable Property Id As Integer

    Public Overridable Property Name As String
End Class

File: CS/SimpleSchedulingExample/Doctor.cs
namespace SimpleSchedulingExample
{

    public class Doctor
    {
        public static Doctor Create()
        {
            return new Doctor();
        }
        public static Doctor Create(int Id, string Name)
        {
            Doctor doctor = Doctor.Create();
            doctor.Id = Id;
            doctor.Name = Name;
            return doctor;
        }

        protected Doctor() { }

        public virtual int Id { get; set; }
        public virtual string Name { get; set; }
    }
}
